apis/AVInput/IAVInput.h

```diff
@@ -0,0 +1,301 @@
+ * following copyright and licenses apply:
+ *
+ * Copyright 2025 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "Module.h"
+
+// @stubgen:include <com/IIteratorType.h>
+
+#define HDMI        0
+#define COMPOSITE   1
+
+namespace WPEFramework {
+    namespace Exchange {
+        /* @json 1.0.0 @text:keep */
+        struct EXTERNAL IAVInput : virtual public Core::IUnknown {
+            enum {
+                ID = ID_AV_INPUT
+            };
+
+            struct MixerLevels {
+                int primaryVolume;
+                int playerVolume;
+            };
+
+            struct InputSignalInfo {
+                int id;
+                string locator;
+                string status;
+            };
+
+            struct InputStatus {
+                InputSignalInfo info;
+                int plane;
+            };
+
+            struct InputVideoMode {
+                int id;
+                string locator;
+                uint32_t width;
+                uint32_t height;
+                bool progressive;
+                uint32_t frameRateN;
+                uint32_t frameRateD;
+            };
+
+            struct GameFeatureStatus {
+                int id;
+                string gameFeature;
+                bool allmMode;
+            };
+
+            struct InputDevice {
+                int id /* @text id */;
+                string locator /* @text locator */;
+                bool connected /* @text connected */;
+            };
+
+            struct ContentInfo {
+                int id;
+                int contentType;
+            };
+
+            using IStringIterator = RPC::IIteratorType<string, RPC::ID_STRINGITERATOR>;
+            using IInputDeviceIterator = RPC::IIteratorType<InputDevice, ID_AV_INPUT_DEVICE_LIST_ITERATOR>;
+
+            // @event
+            struct EXTERNAL IDevicesChangedNotification : virtual public Core::IUnknown {
+                enum {
+                    ID = ID_AV_INPUT_NOTIFICATION_DEVICES_CHANGED
+                };
+
+                // @text onDevicesChanged
+                // @brief Triggered whenever a new HDMI/Composite device is connected to an HDMI/Composite Input
+                // @param devices - in - List of input devices in JSON format
+                virtual void OnDevicesChanged(const string& devices) { }; // Thunder does not currently support iterators as a notification parameter
```

Comment:
Need to use opaque string here instead of an iterator because notification iterators are not supported in Thunder 4.2 (added for 5.x). This was confirmed by Marcel, and the same reason why Suresh used an opaque string for `IAppPackaeManager`. `IUSBMassStorage` notification iterator works because `JUSBMassStorage.h`/`JsonData_USBMassStorage.h` aren’t generated for the plug-in. The problem occurs when the generated event handler attempts to set the `*ParamsData` object to the list parameter.

Reply:
Pete: Add `@opaque`

Reply:
After reviewing PRs for AppPackageManager changes, Marcel recommended they implement their own event handler in order to get around the lack of iterator support in Thunder 4.x. Applied same technique here.